contributors/design-proposals/preserve-order-in-strategic-merge-patch.md

```diff
@@ -0,0 +1,226 @@
+# Preserve Order in Strategic Merge Patch
+
+author: @mengqiy
+
+## Motivation
+
+The order of lists with a merge strategy is important, because an item in the list may depends on an item before it.
+An use case is the environment variables. We don't preserve the order which causes
+issue [40373](https://github.com/kubernetes/kubernetes/issues/40373)
+
+## Proposed Change
+
+Changes are all in strategic merge patch package.
+It will be similar to how we solve the problem of deleting items from a list of primitives.
+
+Always send a parallel list along with the patch for list:
+
+- When patching a list of maps, the parallel list contains a list of items. Each item contains only the merge key.
+- When patching a list of primitives, the parallel list contains a full list from user's config file.
+
+All the items in the server's live list but not in the parallel list will be append to the end of the parallel list.
+The relative order between these appended items are kept.
```

Comment:
Add these as separate sections:

### Behavior when the ordering list contains only some elements

Explain why this would happen.  Explain what happens.

### Behavior when the ordering lists contains elements missing from live value

Explain why this would happen.  Explain what happens.